src/pyop/provider.py

```diff
@@ -180,12 +197,25 @@ def authorize(self, authentication_request,  # type: AuthorizationRequest
                 )
 
             user_claims = self.userinfo.get_claims_for(user_id, requested_claims)
-            response['id_token'] = self._create_signed_id_token(authentication_request['client_id'], sub,
-                                                                user_claims,
-                                                                authentication_request.get('nonce'),
-                                                                authz_code, access_token_value, extra_id_token_claims)
-            logger.debug('issued id_token=%s from requested_claims=%s userinfo=%s extra_claims=%s',
-                         response['id_token'], requested_claims, user_claims, extra_id_token_claims)
+            response['id_token'] = self._create_signed_id_token(
+                authentication_request['client_id'],
+                sub,
+                user_claims,
+                authentication_request.get('nonce'),
+                authz_code,
+                access_token_value,
+                extra_id_token_claims,
+            )
+            logger.debug(
+                'issued id_token=%s from requested_claims=%s userinfo=%s extra_claims=%s',
+                response['id_token'],
+                requested_claims,
+                user_claims,
+                extra_id_token_claims,
+            )
+
+        if self.stateless:
+            del self.userinfo[user_id]
```

Comment:
@smalihaider Why do we need to remove the entry at this point?

There is an assumption that pyop does not modify the user-db. Can we do this on the satosa-frontend level? It is on the satosa-frontend level we also add entries to that db.

Reply:
I agree, change have been pushed.